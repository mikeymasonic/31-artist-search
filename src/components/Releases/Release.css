@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;300&display=swap');

.release {
  height: 20vw;
  width: 30vw;
  display: flex;
  justify-content: center;
  text-align: center;
  margin: 0 3vw 3vw 0;
  padding-top: 3vw;
  border: .2vw solid rgb(133, 220, 226);
  border-radius: 1vw;
  font-size: 1.4vw;
  font-family: 'Montserrat', sans-serif;
}

.release img {
  width: 10vw;
}

.link {
  text-decoration: none;
  color: #f7d89e;
  font-weight: 200;
}

.link:hover {
  color: white;
}

.release:hover {
  background: hotpink;
}

.titleh2 {
  font-size: 2vw;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
  color: #f7d89e;
  font-weight: 200;
}

.titleh3 {
  font-size: 1vw;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
  color: #f7d89e;
  font-weight: 200;
  margin-top: -1vw;
}