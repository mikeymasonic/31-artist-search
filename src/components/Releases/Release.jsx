import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Release = ({
  releaseCover,
  releaseId,
  releaseTitle,
  releaseDate = 'unknown',
  artistName
}) => {
  return (
    <div>
      <Link to={`/songs/${artistName}/${releaseId}/${releaseTitle}`}>
        <img src={releaseCover} />
        <h3>{releaseTitle} - {releaseDate}</h3>
      </Link>
    </div>
  );
};
