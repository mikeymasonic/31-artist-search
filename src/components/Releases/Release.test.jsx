import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release';

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release artistName='Silkworm' releaseCover='dink.png' releaseId='3453lk4' releaseDate='May 4th, 1990' releaseTitle='Firewater' />);
    expect(wrapper).toMatchSnapshot();
  });
});
