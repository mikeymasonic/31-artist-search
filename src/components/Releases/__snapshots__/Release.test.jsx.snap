// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<section
  className="release"
>
  <Link
    className="link"
    to="/songs/Silkworm/3453lk4/Firewater"
  >
    <img
      src="dink.png"
    />
    <h2
      className="titleh2"
    >
      Firewater
    </h2>
     
    <h3
      className="titleh3"
    >
      released: 
      May 4th, 1990
    </h3>
  </Link>
</section>
`;
